doc/march_packages/march_data_collector.rst

```diff
@@ -0,0 +1,184 @@
+        - `Port:`                   8888
+        - `Filter:`                 force pressure cop
+
+More info on this step can be found `here <https://www.moticon.de/doc/science_desktop_software/record/udp/>`_.
+
+3. Lastly use should set the correct launch arguments. You should set pressure_soles to true and if you are not using the
+standard Switch laptop connected to the project March router, you should set the IP-address on which the Moticon desktop app is running.
+
+ROS API
+-------
+
+Nodes
+^^^^^
+*march_data_collector_node* - Responsible for processing IMU data and calculating the jerk.
+
+*com_calculator* - Responsible for calculating the center of mass.
+
+*cp_calculator* - Responsible for calculating the capture points.
+
+*esp_adapter* - Responsible for sending data to the ESP engine.
+
+Subscribed Topics
+^^^^^^^^^^^^^^^^^
+*/march/controller/trajectory/follow_joint_trajectory/feedback* (control_msgs/JointTrajectoryControllerState)
+  Compute the acceleration and jerk.
+
+*/march/imu* (sensor_msgs/Imu)
+  Use IMU data to change the orientation of the tf frames and send to `ESP`.
+
+*/march/pressure_soles* (march_shared_resources/PressureSole)
+  Send pressure sole data to `ESP`.
+
+*/march/imc_states* (march_shared_resources/ImcState)
+  Send iMOTIONCUBE data to `ESP`.
+
+*/march/gait/schedule/goal* (march_shared_resources/GaitActionGoal)
+  Send gait data to `ESP`.
+
+*/march/com_marker* (visualization_msgs/Marker)
+  Send center of mass to `ESP`.
+
+*/march/joint_values* (march_shared_resources/JointValues)
+  Send position, velocity, acceleration and jerk of joints to `ESP`.
+
+*/march/joint_states* (sensor_msgs/JointState)
+  Send actual effort to `ESP`.
+
+*/march/controller/after_limit_joint_command'* (march_shared_resources/AfterLimitJointCommand)
+  Send effort command to `ESP`.
+
+*/march/temperature/\** (sensor_msgs/Temperature)
+  Send temperature command to `ESP` for each joint.
+
+Subscribes to the tf frames trough a `tf2 <http://wiki.ros.org/tf2>`_ buffer.
+
+Published Topics
+^^^^^^^^^^^^^^^^
+*/march/com_marker* (visualization_msgs/Marker)
+  A marker with the CoM point.
+
+*/march/cp_marker_ankle_plate_left* (visualization_msgs/Marker)
+  A marker with the capture point for the left foot.
+
+*/march/cp_marker_ankle_plate_right* (visualization_msgs/Marker)
+  A marker with the capture point for the right foot.
+
+*/march/pressure_soles* (march_shared_resources/PressureSole)
+  A marker with the capture point for the right foot.
+
+*/march/pressure_soles* (march_shared_resources/JointValues)
+  Send the data from the pressure soles.
+
+Publishes into tf via a `tf2 <http://wiki.ros.org/tf2>`_ broadcaster.
+
+Parameters
+^^^^^^^^^^
+*/march/march_data_collector/moticon_ip* (*string*, default: 192.168.8.105)
+  IP-address the Moticon desktop software is running
+*/march/march_data_collector/pressure_soles* (*bool*, default: false)
+  Whether to connect with the pressure soles.
+
+Tutorials
```

Comment:
I think you should add an introduction to explain what ESP does. It listens to ros messages and outputs them to a csv file right? It think it would be nice to explain this and give an example of what you can use it for.

Reply:
Would you make a seperate tuturial on it and reference that one here and only mention the integration here?

Reply:
O I see now that there is already an explanation in the "Event Stream Processing" section. Perhaps it is possible to move the Esp tutorials to the ESP chapter? I think such a structure would be a bit easier to follow. 
Also, I still do not really understand how one would use ESP. I get now how you subscribe to topics and how you create windows, but how do you use the output? How do you finally make the plots you want/get the mean squared error etc. Perhaps it is possible to add an example with a typical usage?